File: jpa-sample/src/main/java/com/example/demo/UserService.java
package com.example.demo;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;

@Service
public class UserService {

    @Autowired
    private UserRepository userRepository;

    @Transactional
    public List<User> findAllUsers() {
        return userRepository.findAll();
    }

    @Transactional
    public void saveUser(User user) {
        userRepository.save(user);
    }
}

File: multi-db/src/main/kotlin/com/example/multidb/service/UserService.kt
package com.example.multidb.service

import com.example.multidb.domain.user.User
import com.example.multidb.domain.user.UserRepository
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Service
class UserService(
    private val userRepository: UserRepository
) {

    @Transactional
    fun getAllUser(): List<User> {
        return userRepository.getAllUser()
    }

    @Transactional
    fun saveUser(name: String, age: Int): User {
        val user = User(name, age)
        userRepository.saveUser(user)
        return user
    }
}
